### ISSUE TITLE
[P1][dogfood] --smart-context returns grade=A/score=100/verdict=SAFE on a file that fails to parse (false green for edit-gating)

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#754]
**Severity:** P1  |  **Surface:** CLI --smart-context on malformed/syntax-error input
_Found by the parallel multi-agent dogfood audit on develop @ 471ac5f5; adversarially re-reproduced and dup-checked by an independent verifier agent._

### Observed
--smart-context returns success=true with health.grade='A', health.score=100.0, health.signal='healthy', agent_summary.verdict='SAFE', edit_risk/risk='safe', recommendation='Grade A, risk: safe', next_step='Make a focused edit, then run the nearby tests and scoped change-impact.' The envelope has NO parse_errors / syntax_error / encoding_warning key. tree-sitter confirms the file does not parse (root_node.has_error=True). For the SAME file, --file-health reports signal='syntax_error', verdict='ERROR', grade='N/A', risk='high', recommendation='File fails to parse...'; --outline reports verdict='WARN' with 'Parse errors detected'. The non-UTF8 file via --smart-context returns grade='F', signal='no_data', verdict='CAUTION' — still never surfaces the syntax_error/encoding signal that file-health emits. Source confirms smart_context_tool.py performs no is_file_parse_broken() check (0 references), unlike file_health_tool.py which short-circuits on it before scoring.

### Expected
--smart-context should detect parse failures the way --file-health/--outline/--code-patterns do (call is_file_parse_broken / short-circuit), and on a non-parsing file downgrade the verdict and surface a parse_errors field or signal='syntax_error' — never report grade=A / score=100 / verdict=SAFE / risk=safe on a file tree-sitter cannot parse. The non-UTF8/binary case should likewise surface an encoding/parse signal rather than a bare no_data. An agent gating an edit on the smart-context verdict must not be told a corrupt file is SAFE to edit.

### Repro
```bash
cd into a TSA checkout (develop). Create a non-parsing Python file:
  mkdir -p _df_tmp && printf 'def broken(\n  x = \n  return\nclass {{{\n' > _df_tmp/syntax_err.py
Run:
  uv run python -m tree_sitter_analyzer --smart-context _df_tmp/syntax_err.py --output-format json
Compare with the sibling tools on the SAME file:
  uv run python -m tree_sitter_analyzer --file-health _df_tmp/syntax_err.py --output-format json
  uv run python -m tree_sitter_analyzer --outline _df_tmp/syntax_err.py --output-format json
Optional non-UTF8 case:
  printf '\xff\xfe\x00\x01broken\xc0\xc1' > _df_tmp/nonutf8.py
  uv run python -m tree_sitter_analyzer --smart-context _df_tmp/nonutf8.py --output-format json
```

### Verifier notes
Re-ran the exact repro in /Users/aisheng.yu/git-private/tsa-dogfood (develop). --smart-context on a file that tree-sitter cannot parse (direct check: root_node.has_error=True) returns success=true, health.grade='A', health.score=100.0, health.signal='healthy', verdict='SAFE', risk='safe', next_step='Make a focused edit...', with NO parse_errors/syntax_error/encoding key. The SAME file via --file-health correctly returns signal='syntax_error', verdict='ERROR', grade='N/A', risk='high'; --outline returns verdict='WARN' 'Parse errors detected'. Non-UTF8 file via --smart-context returns grade='F' signal='no_data' verdict='CAUTION' — never surfaces the syntax_error/encoding signal file-health does. Root cause confirmed in source: file_health_tool.py:205-207 short-circuits on is_file_parse_broken() (utils/parse_validity.py) before scoring; smart_context_tool.py imports only _build_signal and has ZERO references to is_file_parse_broken/parse_validity/syntax_error, calling scorer.score_file() with no parse-validity guard. Flag usage is correct (documented bare 'killer 1-call'), so not a user/flag error. No open issue covers this (only unrelated #733). P1 because it is a false-green on the exact tool an agent uses to gate an edit — an agent would proceed to edit a corrupt file on a SAFE/grade-A verdict.

**Comment by aimasteracc** [comment]
Fixed and merged to develop. See commit in the linked PR.
